airbyte-integrations/connectors/source-chartmogul/source_chartmogul/source.py

```diff
@@ -0,0 +1,94 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+
+from abc import ABC
+from base64 import b64encode
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+
+import requests
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.sources.streams.http.auth import TokenAuthenticator
+
+
+# Basic full refresh stream
+class ChartmogulStream(HttpStream, ABC):
+    url_base = "https://api.chartmogul.com"
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        yield from response.json().get("entries", [])
+
+
+class Customers(ChartmogulStream):
+    primary_key = "uuid"
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        json_response = response.json()
+        if not json_response.get("has_more", False):
+            return None
+
+        return {"page": json_response.get("current_page") + 1}
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        return {"page": 1 if not next_page_token else next_page_token["page"]}
+
+    def path(self, **kwargs) -> str:
+        return "v1/customers"
+
+
+class Activities(ChartmogulStream):
```

Comment:
I think this stream might be a good candidate for an incremental sync mode as you can manipulate the `start_date` and `end_date` parameters. Is there any limitation that prevents you from implementing incremental logic? 

You can do this in another PR if you feel it's too much additional work.

Reply:
It's tricky to make a solid version. Decided not to include in initial PR. Copy-pasting from bootstrap.md:
> One would need to keep track of both UUID and created_at and read stream until `datetime.now()`. Dynamic end date would be necessary since activities can also have a future date. Since data can also be changed retrospectively, a `lookback window` would also be necessary to catch all the changes.